source/common/event/libevent_scheduler.cc

```diff
@@ -15,7 +15,18 @@ TimerPtr LibeventScheduler::createTimer(const TimerCb& cb) {
   return std::make_unique<TimerImpl>(libevent_, cb);
 };
 
-void LibeventScheduler::nonBlockingLoop() { event_base_loop(libevent_.get(), EVLOOP_NONBLOCK); }
+void LibeventScheduler::nonBlockingLoop() {
+#ifdef WIN32
+  // On Windows, EVLOOP_NONBLOCK will cause the libevent event_base_loop to run forever.
+  // This is because libevent only supports level triggering on Windows, and so the write
+  // event callbacks will trigger every time through the loop. Adding EVLOOP_ONCE ensures the
+  // loop will run at most once
+  int flag = EVLOOP_NONBLOCK | EVLOOP_ONCE;
```

Comment:
nit: as long as you are in here can you add const to here and below.

Reply:
Done.